notebooks/sprints/sprint19/530_rsclient_staging.ipynb

```diff
@@ -0,0 +1,520 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "15:17:16.860 [\u001b[32mINFO\u001b[0m] (rs_client.rs_client) POST response vaut: <Response [200]>\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'status': {'started': '36fb0789-9829-4b80-9b6a-fc7f1229568f'}}\n",
+      "\n",
+      "Job ID = 36fb0789-9829-4b80-9b6a-fc7f1229568f\n",
+      "\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "15:17:17.070 [\u001b[32mINFO\u001b[0m] (rs_client.rs_client) POST response vaut: <Response [200]>\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'status': {'started': '45caa458-5886-41bd-aada-f0f64fe643b3'}}\n",
+      "\n",
+      "Job ID = 45caa458-5886-41bd-aada-f0f64fe643b3\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "response_list = []\n",
+    "for items in [items_collection, items_collection_adgs]:\n",
+    "    response_list.append(staging_client.run_staging(items, TEST_COLLECTION))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "589c4f1e",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "TypeError",
+     "evalue": "tuple indices must be integers or slices, not str",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
+      "Cell \u001b[0;32mIn[8], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m response \u001b[38;5;129;01min\u001b[39;00m response_list:\n\u001b[1;32m      3\u001b[0m     \u001b[38;5;28;01mwhile\u001b[39;00m timeout \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[0;32m----> 4\u001b[0m         get_response \u001b[38;5;241m=\u001b[39m staging_client\u001b[38;5;241m.\u001b[39mget_job_status(\u001b[43mresponse\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstatus\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstarted\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[1;32m      5\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m      6\u001b[0m             resp \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mloads(get_response\u001b[38;5;241m.\u001b[39mcontent)\n",
+      "\u001b[0;31mTypeError\u001b[0m: tuple indices must be integers or slices, not str"
+     ]
+    }
+   ],
+   "source": [
+    "timeout = 120\n",
+    "for response in response_list:\n",
+    "    while timeout > 0:\n",
+    "        get_response = staging_client.get_job_status(response[1])\n",
+    "        try:\n",
+    "            resp = json.loads(get_response.content)\n",
```

Comment:
Here it's not user-friendly when your wrapped function returns an http response that the end-user has to load as a json and parse. This should be done ("hidden from the end-user") in your wrapped function. It could return an enum that simply gives the status value. See the existing enum EDownloadStatus. Maybe you can re-user/update it since it's deprecated. Please discuss this with Vincent.